### PR TITLE
feat: Added UID to workflows instead of username

### PR DESCRIPTION
**Pull request by vanshBhatia-A4k9**
## Proposed changes

workflows now show UID instead of username as a unique identifier

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [ ] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the commit for DCO to be passed.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

### CODE REVIEW

File: litmus-portal/graphql-server/pkg/cluster/handler/handler.go [hunk 1]
```diff
@@ -277,7 +277,7 @@ func SendRequestToSubscriber(subscriberRequest clusterOps.SubscriberRequests, r
 			Namespace:    subscriberRequest.Namespace,
 			RequestType:  subscriberRequest.RequestType,
 			ExternalData: subscriberRequest.ExternalData,
-			Username:     subscriberRequest.Username,
+			UID:          subscriberRequest.Username,
```
Comment (Jonsy13) [inline litmus-portal/graphql-server/pkg/cluster/handler/handler.go:280]: I thin, this should be UID
```suggestion
			UID:          subscriberRequest.UID,
```

File: litmus-portal/graphql-server/pkg/chaos-workflow/ops/operations.go [hunk 2]
```diff
@@ -248,7 +248,7 @@ func SendWorkflowToSubscriber(workflow *model.ChaosWorkFlowRequest, username *st
 		ClusterID:    workflow.ClusterID,
 		Namespace:    workflowNamespace,
 		ExternalData: externalData,
-		Username:     username,
+		Username:     uid,
```
Comment (Jonsy13) [inline litmus-portal/graphql-server/pkg/chaos-workflow/ops/operations.go:251]: This should be UID
```suggestion
		UID:     uid,
```

**Review by Jonsy13 — CHANGES_REQUESTED** [review]

**Comment by gdsoumya** [comment]
UIDs will probably solve the current issue but this also makes it less auditable unless we let the users/admins query accounts using uids. I don't remember having such UI currently but maybe I am wrong @vanshBhatia-A4k9

**Comment by amityt** [comment]
Hi @brnck Thanks for reaching out! After some discussion with the team, we have decided to fix the issue from the DEX server itself, also we'll add some additional checks in the auth server to prevent any special characters in the username. It should be available by 2.13 as it is a larger change and will take some time to test the same.

@gdsoumya we'll be adding the user metadata that contains the username as well in the schema, but we'll only pass the userID in the manifest.

**Comment by bbarin** [comment]
As @gdsoumya mentioned, changed to UUID will solve but breaks the auditing, which is very important. I would rather base64 the username to circumvent this issue.

**Comment by Saranya-jena** [comment]
These changes have been added as part of 3.0

**Pull request closed without merging**